test/e2e/framework.go

```diff
@@ -525,25 +525,40 @@ func (data *TestData) collectExternalInfo() error {
 		}
 	}
 
-	subnets := strings.Split(testOptions.vlanSubnets, ",")
-	for _, subnet := range subnets {
-		if subnet == "" {
+	vlanSubnetsList := strings.Split(testOptions.vlanSubnets, ",")
+	for _, vlanSubnets := range vlanSubnetsList {
+		if vlanSubnets == "" {
 			continue
 		}
-		gatewayIP, _, err := net.ParseCIDR(subnet)
+		parts := strings.Split(vlanSubnets, "=")
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid vlan subnet format: %s", vlanSubnets)
+		}
```

Comment:
I think this handling is wrong given the latest change? How can this code handle input like `10=172.100.10.1/24,fd00:172:100:10::1/96`?